sale_order_product_assortment/models/sale_order.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2020 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    whitelist_product_ids = fields.Many2many(
+        comodel_name="product.product",
+        string="Whitelisted Products",
+        compute="_compute_product_assortment_ids",
+    )
+    blacklist_product_ids = fields.Many2many(
+        comodel_name="product.product",
+        string="Blacklisted Products",
+        compute="_compute_product_assortment_ids",
+    )
+    has_whitelist = fields.Boolean(compute="_compute_product_assortment_ids")
+    has_blacklist = fields.Boolean(compute="_compute_product_assortment_ids")
+
+    @api.depends("partner_id")
+    def _compute_product_assortment_ids(self):
+        # If we don't initialize the fields we get an error with NewId
+        self.whitelist_product_ids = self.env["product.product"]
```

Comment:
Don't you loop over the recorset?

Reply:
In this case I want to initialize the field as a empty recordset. I don't know if this is the correct way

Reply:
If that's what you want, that seems like the correct way, according to the [documentation](https://www.odoo.com/documentation/13.0/reference/orm.html#environment).
However, I don't really understand the logic... AFAICS, you are reassigning it [bellow](https://www.odoo.com/documentation/13.0/reference/orm.html#environment), right? What was the error with `NewId` you refer to in the comment?

Reply:
But what about later?

Reply:
I just want to have records if a partner is defined... so here you have were we add the records https://github.com/OCA/sale-workflow/pull/1407/files#diff-d857107550de88517b0f4cb32ae7770a25a630dc254b97191e7f71f262183089R52-R57

Reply:
Can you write over multiple records in v13 with a direct assignment?

Reply:
@joao-p-marques that's the error that i catch if I don't use this...

Reply:
Ok, does it give any more information in the logs?

Reply:
@joao-p-marques the logs just give me this info `2021-01-04 09:43:48,203 1 WARNING devel odoo.http: ('sale.order(<NewId 0x7f7c50380ef0>,).whitelist_product_ids', None)`

Reply:
Ok. Your latest push also cleared it for me :smile: Good job!